### PR TITLE
CEPHSTORA-154 Add rpco-master-rc integration test

### PR DESCRIPTION
**Pull request by andymcc**

### CODE REVIEW

**Comment by andymcc** [comment]
This will need https://github.com/rcbops/rpc-ceph/pull/155 to be merged for the rpco_pike job to work. rpco_master is unreliable so we would more just use rpco_pike.

Additionally, there are gating failures in rpco-pike right now that are unrelated to ceph, but this PR is ready to go.

File: gating/pre_merge_test/run [hunk 1]
```diff
@@ -133,35 +159,36 @@ export ANSIBLE_INVENTORY="${CLONE_DIR}/tests/inventory"
...
 # RPC-O has existing lxc roles that will conflict
-if [ "${RE_JOB_SCENARIO}" != "rpco_newton" ]; then
+if [ "${RE_JOB_SCENARIO}" != "rpco_newton" ] && [ "${RE_JOB_SCENARIO}" != "rpco_master" ] && [ "${RE_JOB_SCENARIO}" != "rpco_pike" ]; then
   ${ANSIBLE_BINARY} playbooks/git-clone-repos.yml \
                    -i ${ANSIBLE_INVENTORY} \
                    -e role_file=../tests/ansible-role-test-requirements.yml
 fi
 ## Clone RPC-O directory into /opt
-if [ "${RE_JOB_SCENARIO}" == "rpco_newton" ]; then
+if [ "${RE_JOB_SCENARIO}" == "rpco_newton" ] || [ "${RE_JOB_SCENARIO}" == "rpco_master" ] || [ "${RE_JOB_SCENARIO}" == "rpco_pike" ]; then
```
Comment (mvollman) [inline gating/pre_merge_test/run]: Could this be `if [[ "${RE_JOB_SCENARIO}" =~ ]]`?

File: gating/pre_merge_test/run [hunk 2]
```diff
@@ -70,10 +69,26 @@ function _set_rpco_vars {
 }
 
 function _configure_rpco {
+  if [ "${RE_JOB_SCENARIO}" = "rpco_newton" ]; then
```
Comment (mvollman) [inline gating/pre_merge_test/run]: Could we remove the else if block and just do:
```
export RPCO_VERSION="${RE_JOB_SCENARIO#rpco_}-rc"
export OSA_DIR="${RPCO_DIR}/openstack-ansible"
export SECRETS_FILE="/etc/openstack_deploy/user_osa_secrets.yml"
export CINDER_ENVD="inventory/env.d/cinder.yml"
```

File: gating/pre_merge_test/run [hunk 3]
```diff
@@ -82,13 +97,24 @@ function _configure_rpco {
...
     ## Bootstrap RPC-O's ansible
-    bash scripts/bootstrap-ansible.sh
-    cp ${CLONE_DIR}/tests/rpco_vars/user_rpc_ceph_vars.yml /etc/openstack_deploy/user_rpc_ceph_vars.yml
-    bash scripts/bootstrap-aio.sh
+    if [ "${RE_JOB_SCENARIO}" = "rpco_newton" ]; then
+      bash scripts/bootstrap-ansible.sh
+      cp ${CLONE_DIR}/tests/rpco_vars/user_rpc_ceph_vars.yml /etc/openstack_deploy/user_rpc_ceph_vars.yml
+      bash scripts/bootstrap-aio.sh
+    fi
+    if [ "${RE_JOB_SCENARIO}" = "rpco_master" ] || [ "${RE_JOB_SCENARIO}" = "rpco_pike" ]; then
```
Comment (mvollman) [inline gating/pre_merge_test/run]: Can we drop this conditional and just assume this is the RPCO method when NOT newton moving forward?

File: gating/pre_merge_test/run [hunk 4]
```diff
@@ -133,35 +159,36 @@ export ANSIBLE_INVENTORY="${CLONE_DIR}/tests/inventory"
...
   git clone https://github.com/openstack/openstack-ansible-tests -b stable/pike tests/common
 fi
-_get_data_disk
 ${ANSIBLE_BINARY} tests/setup-ceph-aio.yml \
                  -i ${ANSIBLE_INVENTORY} \
-                 -e @tests/test-vars.yml \
-                 ${ANSIBLE_ADDITIONAL_VARS} ${BOOTSTRAP_OPTS}
-if [ "${RE_JOB_SCENARIO}" == "rpco_newton" ]; then
+                 -e @tests/test-vars.yml ${CEPH_BOOTSTRAP_OPTS}
+if [ "${RE_JOB_SCENARIO}" == "rpco_newton" ] || [ "${RE_JOB_SCENARIO}" == "rpco_master" ] || [ "${RE_JOB_SCENARIO}" == "rpco_pike" ]; then
```
Comment (mvollman) [inline gating/pre_merge_test/run]: Same thing here with matching conditional

File: gating/pre_merge_test/run [hunk 5]
```diff
@@ -133,35 +159,36 @@ export ANSIBLE_INVENTORY="${CLONE_DIR}/tests/inventory"
 export ANSIBLE_PARAMETERS="-e @${CLONE_DIR}/tests/test-vars.yml -e @${RPC_MAAS_DIR}/tests/user_rpcm_secrets.yml"
 export ANSIBLE_BINARY="${ANSIBLE_BINARY:-ceph-ansible-playbook}"
 bash scripts/bootstrap-ansible.sh
+if [ "${RE_JOB_SCENARIO}" != "rpco_newton" ]; then
+  _get_data_disk
+fi
 # Clone the test repos when not integrating with RPC-O
 # RPC-O has existing lxc roles that will conflict
-if [ "${RE_JOB_SCENARIO}" != "rpco_newton" ]; then
+if [ "${RE_JOB_SCENARIO}" != "rpco_newton" ] && [ "${RE_JOB_SCENARIO}" != "rpco_master" ] && [ "${RE_JOB_SCENARIO}" != "rpco_pike" ]; then
```
Comment (mvollman) [inline gating/pre_merge_test/run]: Same thing here with matching conditional

**Pull request merged**